letsencrypt/display/completer.py

```diff
@@ -0,0 +1,61 @@
+"""Provides tab completion when prompting users for a path."""
+import glob
+# readline module is not available on all systems
+try:
+    import readline
+except ImportError:
+    import letsencrypt.display.dummy_readline as readline
+
+
+class Completer(object):
+    """Provides tab completion when prompting users for a path.
+
+    This class is meant to be used with readline to provide tab
+    completion for users entering paths. The complete method can be
+    passed to readline.set_completer directly, however, this function
+    works best as a context manager. For example:
+
+    with Completer():
+        raw_input()
+
+    In this example, tab completion will be available during the call to
+    raw_input above, however, readline will be restored to its previous
+    state when exiting the body of the with statement.
+
+    """
+
+    def __init__(self):
+        self._completer = self._delims = self._iter = None
+
+    def complete(self, text, state):
+        """Provides path completion for use with readline.
+
+        :param str text: text to offer completions for
+        :param int state: which completion to return
+
+        :returns: possible completion for text or ``None`` if all
+            completions have been returned
+        :rtype: str
+
+        """
+        if state == 0:
+            self._iter = glob.iglob(text + '*')
+        return next(self._iter, None)
```

Comment:
Strictly speaking, this assumes that the completes get requested in order, right? If a called requested 7 and then 5, it would not actually go into the glob data and get item 7 and then item 5, but just return the next two in the currently-ongoing sequential order.

(I don't think this is a practical problem in terms of how readline will call this.)